src/main/java/service/imp/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,28 @@
+package service.imp;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import java.util.Map;
+import java.util.Set;
+import service.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String COLUMNS_NAME = "fruit,quantity";
+    private FruitDao fruitDao;
+
+    public ReportGeneratorImpl() {
+        this.fruitDao = new FruitDaoImpl();
+    }
```

Comment:
The direct instantiation of `FruitDaoImpl` within the `ReportGeneratorImpl` constructor makes it less independent and violates the Dependency Inversion Principle. Consider passing `FruitDao` as a constructor parameter.